### ISSUE TITLE
ray.get() hangs if you call it on a actor task that calls sys.exit()

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#1049]
I was trying to kill actor processes to release GPU resources by having a method that would raise SystemExit. However, ray.get() hangs forever in this case. I see in the console:

```
Disconnecting client on fd 42                                                                              
Error in atexit._run_exitfuncs:                                                                            
Traceback (most recent call last):                                                                         
  File "/home/ubuntu/.local/lib/python3.6/site-packages/ray-0.2.0-py3.6-linux-x86_64.egg/ray/worker.py", li
ne 1358, in cleanup                                                                                        
A worker died or was killed while executing a task.                                                        
                                                                                                           
  You can inspect errors by running                                                                        
                                                                                                           
      ray.error_info()                                                                                     
                                                                                                           
  If this driver is hanging, start a new one with                                                          
                                                                                                           
      ray.init(redis_address="127.0.0.1:25614")                                                            
                                                                                                           
    worker.plasma_client.disconnect()                                                                      
  File "/home/ubuntu/.local/lib/python3.6/site-packages/ray-0.2.0-py3.6-linux-x86_64.egg/ray/worker.py", li
ne 865, in exit                                                                                            
    sys.exit(0)                                                                                            
SystemExit: 0                                                                                              
```

If you interrupt the driver, you see that it's hanging in the `ray.get()` for the stop method.
```
Traceback (most recent call last):
  File "./rllib-eval.py", line 261, in <module>
    runner.process_events()
  File "./rllib-eval.py", line 200, in process_events
    exp.stop()
  File "./rllib-eval.py", line 66, in stop
    ray.get(self.agent.stop.remote())
```

**Comment by robertnishihara** [comment]
Good point, I just verified that this also happens for remote functions (though in the remote function case the task gets resubmitted over and over by the local scheduler).

I think what we want to do here is the following. The local scheduler detects that the worker died, it looks at what task the worker was executing and then there are a couple options.
1. It can put an object representing a failed task in the object store for the objects that were supposed to be created by the failed task. Whoever calls `ray.get` on that object will retrieve the object and reraise the exception. This is the normal approach we use when tasks raise exceptions.
2. It can stick a key in Redis saying that the task failed (this is easy to implement), but then whoever is waiting for the objects will have to check Redis, which is pretty inconvenient (that is it's not clear what the right way to do this is when you're calling `ray.get` on one million objects).

**Comment by robertnishihara** [comment]
Fixed by #1224.
